hudi-hive-sync/src/main/java/org/apache/hudi/hive/HoodieHiveClient.java

```diff
@@ -198,7 +198,8 @@ private String getPartitionClause(String partition) {
     for (String partition : partitions) {
       String partitionClause = getPartitionClause(partition);
       Path partitionPath = FSUtils.getPartitionPath(syncConfig.basePath, partition);
-      String fullPartitionPath = partitionPath.toUri().getScheme().equals(StorageSchemes.HDFS.getScheme())
+      String partitionScheme = partitionPath.toUri().getScheme();
+      String fullPartitionPath = StorageSchemes.HDFS.getScheme().equals(partitionScheme)
```

Comment:
How does this work for non hdfs FS like cloud stores ?

Reply:
The original code itself does not seem to support non HDFS sources. Are non HDFS cloud stores even supported by Hive?

This fix has only changed the order of comparsion as getScheme() may return NULL when there is no scheme prefix like in the unit-tests. The boolean result of the comparison remains the same.

Reply:
No, I think it does support cloud stores. This special handling was part of https://jira.apache.org/jira/browse/HUDI-325  to get correct behavior for HDFS  for alter partitions.  This should be fine.